### DISCUSSION TITLE
need help: plugin to create G0 gcode from current x,y,z position from ControlPanel + Camera

### DISCUSSION THREAD

**Discussion by beta-tester** [vlachoudis/bCNC#1706] [General]
hi,
i want to use the Camera and jogging in the ControlPanel to goto different positons and create G0 gcode from X,Y,Z position of interest by press a keyboard key and/or a click to a button in the GUI.

i wondering if there is already the poosibility to do that, but i cant find it.
i tried to find a plugin that is reading the current CNC spindle position to append its positon to G0 gcode 
something like `block.append(CNC.grapid(X,Y,Z))` but i don't need a full new block, i just need a new line.
but i all plugins i found are only on the CAMPanel but not on the ControlPanel and all plugins open a parameter GUI you have to interact with - i need a parameter less solution that read the current X,Y,Z position from the CNC and put it with out further interaction to the gcode..

can somebody give me a hint how to add a plugin to the ControlPanel, how to ready the current X,Y,Z position from the CNC and how to append gcode line (without generating a full new gcode block).

**Comment by MARIOBASZ** [comment]
I fail to understand the usefulness of what you are looking for.
I think you can see the code used in Probe - Probe - Record , it looks very similar to what you intended.

**Reply by beta-tester** [reply]
i want to inspect many specific points on a device with the camera.
so first i need to take the points from a good device. to input them i want to use bCNC to precisely navigate to the points of interest (points to inspect).
so i navigate to the points and i want to press a button on each point of interest to take the X,Y,Z coordinates to a gcode file.
then i can later "play" that recorded coordinates to inspect other devices of the same type to see if they are good as well.
(take camera pictures of each inspection point for quality documentation/evidence)

the Probe-Probe-Record is that a plugin? i cant see it in the plugins folder.

**Reply by Harvie** [reply]
Probe and Record are not plugins, they are direct feature of bCNC

**Comment by MARIOBASZ** [comment]
![586109E2-E326-4832-B294-A45109F88B0E](https://user-images.githubusercontent.com/11743403/171843208-05c21542-f579-4092-926f-765d414b8c6f.jpeg)

**Reply by beta-tester** [reply]
oops, stupid me - what i want is already there.
yes, the RAPID button on the ProbePage is what i am looking for.

thank you.
maybe i have to add a keyboard shortcut - or does it has already one?

**Reply by beta-tester** [reply]
ok, i found the names required for the shortcuts
`RRAPID` is the record rapid button
`RFINISH` is the record finish button

**Comment by beta-tester** [comment]
is it somehow possible to show the CNC spindle position always at the center of the view and set the scaling to a spacific value so that the camera scale fits with the view scale and the spindle position to the camera position?
at the moment the camera is at the center, but the position of the coordinate system in the view is fix and the spindle is moving in the view and i don't see the ability to set a specific scale to the view so that one mm on the camera is one mm in the view
![bcnc-canvas](https://user-images.githubusercontent.com/4750719/171904763-9116b030-d3bc-4926-b337-fa43865bedd4.png)

... sorry i don't know how to explain it better.
